org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/DocumentProvider.xtend

```diff
@@ -65,26 +66,43 @@ class DocumentProvider {
 
 	def boolean delete(String resourcePath) {
 		val file = getWorkspaceFile(resourcePath)
-		if (!file.exists) {
+		if(file.exists) {
+			val deleted = FileUtils.deleteQuietly(file)
+			if(deleted) {
+				commit(file)
+			}
+			return deleted
+		} else {
 			throw new FileNotFoundException(resourcePath)
 		}
-		return FileUtils.deleteQuietly(file)
 	}
 
-	private def void write(File file, String content) {
+	private def void write(
+		File file,
+		String content
+	) {
 		Files.asCharSink(file, UTF_8).write(content)
 		commit(file)
 	}
 
 	private def void commit(File file) {
-		val workspace = workspaceProvider.workspace
 		val git = gitProvider.git
-		val filePattern = workspace.toPath.relativize(file.toPath).toString
-		git.add.addFilepattern(filePattern).call
+		git.stage(file)
 		git.commit.setMessage("bla").call
 		updateGit
 	}
 
+	private def void stage(Git git, File file) {
+		val workspace = workspaceProvider.workspace
+		val filePattern = workspace.toPath.relativize(file.toPath).toString
+		if(file.exists) {
```

Comment:
1. I don't think that's valid Xtend, it would have to be something like
`val gitCommand = if (file.exists) git.add else git.rm`. Right…?
2. Unfortunately, while `AddCommand` and `RmCommand` share a common base class, they both declare an individual method `addFilepattern(…)`. They have matching signatures, but the type of `gitCommand` is inferred to be that of the base class `GitCommand`, which does not have such a method, at all. In fact, I first tried to write it more like the way you propose, but due to this flaw in the API, it does not work that way.